src/xregexp.js

```diff
@@ -791,21 +793,28 @@ XRegExp.escape = function(str) {
 XRegExp.exec = function(str, regex, pos, sticky) {
     var cacheKey = 'g',
         addY = false,
+        fakeY = false,
         match,
         r2;
 
     addY = hasNativeY && !!(sticky || (regex.sticky && sticky !== false));
     if (addY) {
         cacheKey += 'y';
+    } else if (sticky) {
+        cacheKey += 'fakeY'
+        fakeY = true
```

Comment:
Oh, yeah, sorry, I don't use them in my own code and forgot to check what you're using. I'll push a rebased, fixed version.